interview-questions.md

```diff
@@ -8,36 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: We first need to set up the realtionship. the foreign key must be part of the student model and its going to be named cohort_id. We need to create a migration file to to add the column then migrate it.
 
-Researched answer:
+Researched answer:Generate a migration in Rails to add the foreign key column. Inside the migration file, we need to add the code to create the foreign key column.After generating the migration and adding the column definition, you can run the migration.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The create and update. The reasons is we are creating and updating a new instance and we set up requiered param that the user must provide
 
-Researched answer:
+Researched answer:POST route: This route is used for creating a new resource on the server.PUT/PATCH route: These routes are used for updating an existing resource on the server.GET route with query parameters: While not required, it is common to include query parameters in a GET request to retrieve specific data or filter the results. DELETE route with path parameters: The DELETE route is used for deleting a specific resource from the server.These routes require parameters to provide the necessary information to perform the desired action on the server. 
```

Comment:
The answer would be show, edit, update, delete! Remember that GET POST PUT PATCH DELETE are HTTP verbs.